Tools/Algorithms/stencil.py

```diff
@@ -417,41 +446,87 @@ def run_main(dx, dy, dz, dt, nox, noy, noz, gamma=1., galilean=False, path='.',
         + '\nis shaded to help you identify a suitable number of ghost cells.')
     print('\nFor a nodal simulation, choose:')
     print(f'- between {gcmin_x_nodal} and {gcmax_x_nodal} ghost cells along x')
-    print(f'- between {gcmin_y_nodal} and {gcmax_y_nodal} ghost cells along y')
-    print(f'- between {gcmin_z_nodal} and {gcmax_z_nodal} ghost cells along z')
+    if dims == 3:
+        print(f'- between {gcmin_y_nodal} and {gcmax_y_nodal} ghost cells along y')
+    if dims > 1:
+        print(f'- between {gcmin_z_nodal} and {gcmax_z_nodal} ghost cells along z')
     print('\nFor a staggered or hybrid simulation, choose:')
     print(f'- between {gcmin_x_stagg} and {gcmax_x_stagg} ghost cells along x')
-    print(f'- between {gcmin_y_stagg} and {gcmax_y_stagg} ghost cells along y')
-    print(f'- between {gcmin_z_stagg} and {gcmax_z_stagg} ghost cells along z')
+    if dims == 3:
+        print(f'- between {gcmin_y_stagg} and {gcmax_y_stagg} ghost cells along y')
+    if dims > 1:
+        print(f'- between {gcmin_z_stagg} and {gcmax_z_stagg} ghost cells along z')
     print()
 
     return
 
 if __name__ == '__main__':
 
-    # --
-    # User can modify these input parameters
-    # --
-    # Cell size
-    dx = 1e-06
-    dy = 1e-06
-    dz = 2e-06
-    # Time step
-    dt = 1e-14
-    # Spectral order
-    nox = 8
-    noy = 8
-    noz = 16
-    # Lorentz boost
-    gamma = 30.
+    # Parse path to input file from command line
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--path')
+    args = parser.parse_args()
+    input_file = args.path
+
+    # Parse input file
+    input_dict = dict()
+    for line in open(input_file):
+        sline = line.split('=')
+        # skip lines that are commented out or blank
+        skip_line = sline[0].startswith('#') or sline[0].startswith('\n')
+        if not skip_line:
+            key = sline[0].strip()
+            val = sline[1].split()
+            input_dict[key] = val
+            # The value corresponding to a given key of input_dict is a list
+            # of strings, from which we remove any leftover comments
+            for w in input_dict[key]:
```

Comment:
I would remove the comments from `val` first, then add it to the dictionary. I think that would be cleaner.

Reply:
I'm thinking that maybe keeping a list of length 1 for one-dimensional input values makes it easier to make the script handle 1D without further changes. For example, a line like this 
```py
    nx = np.array([int(w) for w in input_dict['amr.n_cell']])
```
probably would not work if `input_dict['amr.n_cell']` were a single integer instead of a list of integers of length 1, which would be the case for 1D input files.